react-popup/my-app/src/App.tsx

```diff
@@ -0,0 +1,37 @@
+import { useRef, useState } from 'react';
+import './App.css';
+import { Popup } from './Popup';
+
+export default function App() {
+  const [poppedUp, setPoppedUp] = useState(false);
+  const theButton = useRef(null).current;
+  const menuItems = ['this', 'that', 'the other'];
+  function handleClose() {
+    setPoppedUp(false);
+  }
+  return (
+    <>
+      <button
+        ref={theButton}
+        className="hover:bg-[#aaa]"
+        onClick={() => setPoppedUp(!poppedUp)}>
+        Pop up!
+      </button>
+      <p>Some random text</p>
+      {poppedUp && (
```

Comment:
You can conditionally render this if you'd like, but it's not necessary because it removes itself from the DOM if it's not popped up. (See Popup.tsx line 24, where it returns null if `poppedUp` is false.)